### PR TITLE
Reduce old object sample event recordings size

### PR DESCRIPTION
**Pull request by flodav**

### CODE REVIEW

File: src/hotspot/share/jfr/leakprofiler/checkpoint/objectSampleCheckpoint.cpp [hunk 1]
```diff
@@ -197,6 +197,62 @@ static bool stack_trace_precondition(const ObjectSample* sample) {
   return sample->has_stack_trace_id() && !sample->is_dead();
 }
 
+class StackTraceChunkWriter {
```
Comment (flodav) [inline src/hotspot/share/jfr/leakprofiler/checkpoint/objectSampleCheckpoint.cpp:200]: `StackTraceChunkWriter` writes Stack traces to the given `JfrChunkWriter` using the leak profiler `JfrStackTraceRepository`.
This is essentially the  same logic as in `StackTraceBlobInstaller`.

File: src/hotspot/share/jfr/recorder/service/jfrRecorderService.cpp [hunk 2]
```diff
@@ -264,6 +264,42 @@ static u4 invoke_with_flush_event(Functor& f) {
   return elements;
 }
 
+class ObjectSamplerStackTraceRepository : public StackObj {
```
Comment (flodav) [inline src/hotspot/share/jfr/recorder/service/jfrRecorderService.cpp:267]: This class allows `JfrRecorder` to write stack traces from the leak profiler to the current chunk.

File: src/hotspot/share/jfr/recorder/stacktrace/jfrStackTraceRepository.hpp [hunk 3]
```diff
@@ -52,6 +52,8 @@ class JfrStackTraceRepository : public JfrCHeapObj {
   JfrStackTraceRepository();
   static JfrStackTraceRepository* create();
   static void destroy();
+  static JfrStackTraceRepository* create_leak_profiler();
+  static void destroy_leak_profiler();
```
Comment (jbachorik) [inline src/hotspot/share/jfr/recorder/stacktrace/jfrStackTraceRepository.hpp:58]: Does it make sense to have 2 distinct `destroy()` methods here?
Couldn't the original `destroy()` take care of both the ordinary and leakp instance?

**Review by jbachorik — COMMENTED** [review]
From my PoV this looks pretty good.
The final PR should be open against the openjdk master (the original repo, not the DataDog fork).

File: src/hotspot/share/jfr/recorder/stacktrace/jfrStackTraceRepository.hpp (hunk 3 above)
Comment (flodav) [reply src/hotspot/share/jfr/recorder/stacktrace/jfrStackTraceRepository.hpp:58]: It is possible to combine both `destroy()` but I think there is an underlying convention to have 1 object - 1 constructor - 1 destructor. It seems to be the case everywhere in JFR code.

Comment (jbachorik) [reply src/hotspot/share/jfr/recorder/stacktrace/jfrStackTraceRepository.hpp:58]: 👍

**Pull request closed without merging**